hack/generator/pkg/astmodel/armconversion/shared.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package armconversion
+
+import (
+	"fmt"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"go/ast"
+	"sync"
+)
+
+type conversionBuilder struct {
+	receiverIdent              *ast.Ident
+	receiverTypeExpr           ast.Expr
+	armTypeIdent               *ast.Ident
+	codeGenerationContext      *astmodel.CodeGenerationContext
+	idFactory                  astmodel.IdentifierFactory
+	isResource                 bool
+	methodName                 string
+	kubeType                   *astmodel.ObjectType
+	armType                    *astmodel.ObjectType
+	propertyConversionHandlers []propertyConversionHandler
+}
+
+func (builder conversionBuilder) propertyConversionHandler(
+	toProp *astmodel.PropertyDefinition,
+	fromType *astmodel.ObjectType) []ast.Stmt {
+
+	for _, conversionHandler := range builder.propertyConversionHandlers {
+		matched, matchedProperty := conversionHandler.finder(toProp, fromType)
```

Comment:
Yeah I agree it would probably be weird to have a situation where you have a type which you are assigning to, it has 7 properties, and you don't want to fill one of them out.

I guess it could happen if we have a field that was read-only or something, but at least in theory that shouldn't be on the spec then (since you can't set it)

Reply:
Did this